mmv1/products/dialogflowcx/Agent.yaml

```diff
@@ -111,3 +112,113 @@ properties:
+          * Agent level
+          * Flow level
+          * Page level
+          * Parameter level
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: 'enabled'
+            description: |
+              If true, incoming audio is processed for DTMF (dual tone multi frequency) events. For example, if the caller presses a button on their telephone keypad and DTMF processing is enabled, Dialogflow will detect the event (e.g. a "3" was pressed) in the incoming audio and pass the event to the bot to drive business logic (e.g. when 3 is pressed, return the account balance).
+          - !ruby/object:Api::Type::Integer
+            name: 'maxDigits'
+            description: |
+              Max length of DTMF digits.
+          - !ruby/object:Api::Type::String
+            name: 'finishDigit'
+            description: |
+              The digit that terminates a DTMF digit sequence.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'loggingSettings'
+        send_empty_value: true
+        allow_empty_object: true
+        description: |
+          Settings for logging. Settings for Dialogflow History, Contact Center messages, StackDriver logs, and speech logging. Exposed at the following levels:
+          * Agent level
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: 'enableStackdriverLogging'
+            send_empty_value: true
+            description: |
+              If true, StackDriver logging is currently enabled.
+          - !ruby/object:Api::Type::Boolean
+            name: 'enableInteractionLogging'
+            send_empty_value: true
+            description: |
+              If true, DF Interaction logging is currently enabled.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'gitIntegrationSettings'
+    description: |
+      Git integration settings for this agent.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'githubSettings'
+        description: |
+          Settings of integration with GitHub.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'displayName'
+            description: |
+              The unique repository display name for the GitHub repository.
+          - !ruby/object:Api::Type::String
+            name: 'repositoryUri'
+            description: |
+              The GitHub repository URI related to the agent.
+          - !ruby/object:Api::Type::String
+            name: 'trackingBranch'
+            description: |
+              The branch of the GitHub repository tracked for this agent.
+          - !ruby/object:Api::Type::String
+            name: 'accessToken'
+            description: |
+              The access token used to authenticate the access to the GitHub repository.
+            sensitive: true
+            ignore_read: true
```

Comment:
Discussed in chat- we need to use a custom flatten, similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/custom_flatten/filestore_instance_networks_reserved_ip_range.go.erb

`ignore_read` only works as intended on top level fields

Reply:
Gotcha. I did a thing and it seemed to work in local testing 🤞, will resolve this when the build is green.